bootstrap/internal/controller/agentbootstrapconfig_controller.go

```diff
@@ -483,9 +496,14 @@ func (r *AgentBootstrapConfigReconciler) setMetal3MachineTemplateImage(
 
 	if machineTpl.Spec.Template.Spec.Image.URL != config.Status.ISODownloadURL ||
 		machineTpl.Spec.Template.Spec.Image.DiskFormat != &liveIsoFormat {
+		patchHelper, err := patch.NewHelper(machineTpl, r.Client)
 		machineTpl.Spec.Template.Spec.Image.URL = config.Status.ISODownloadURL
 		machineTpl.Spec.Template.Spec.Image.DiskFormat = &liveIsoFormat
-		if err := r.Client.Update(ctx, machineTpl); err != nil {
+		if err != nil {
+			log.Error(err, "couldn't create patch helper for machineTpl")
```

Comment:
Should we do something here? If there's an error creating the patchHelper then we probably shouldn't run this line right? https://github.com/openshift-assisted/cluster-api-agent/pull/50/files#diff-f7f355380e3be4093bf66be019c03d3592f6d6b366191d2fc800a57eb1599241R506